render/detailed_node.go

```diff
@@ -153,18 +151,27 @@ func OriginTable(r report.Report, originID string) (Table, bool) {
 	return Table{}, false
 }
 
-func connectionDetailsRows(endpointTopology report.Topology, originID string, nmd report.NodeMetadata) []Row {
+func connectionDetailsRows(topology report.Topology, originID string) []Row {
 	rows := []Row{}
-	local := fmt.Sprintf("%s:%s", nmd.Metadata[endpoint.Addr], nmd.Metadata[endpoint.Port])
-	adjacencies := endpointTopology.Adjacency[report.MakeAdjacencyID(originID)]
-	sort.Strings(adjacencies)
-	for _, adj := range adjacencies {
-		if _, address, port, ok := report.ParseEndpointNodeID(adj); ok {
-			rows = append(rows, Row{
-				Key:        local,
-				ValueMajor: fmt.Sprintf("%s:%s", address, port),
-			})
+
+	labeler := func(nodeID string) string {
+		if _, addr, port, ok := report.ParseEndpointNodeID(nodeID); ok {
+			return fmt.Sprintf("%s:%s", addr, port)
+		}
+		if _, addr, ok := report.ParseAddressNodeID(nodeID); ok {
+			return addr
 		}
+		return ""
+	}
+
+	local := labeler(originID)
+	adjacencies := topology.Adjacency[report.MakeAdjacencyID(originID)]
+	sort.Strings(adjacencies)
+	for _, nodeID := range adjacencies {
+		rows = append(rows, Row{
```

Comment:
`labeler` can return an empty string. We shouldn't add a row when this happens. And, if it *can't* happen then I think we should make it explicit (e.g. by logging it).